examples/pytorch/dimenet/qm9.py

```diff
@@ -0,0 +1,218 @@
+    Raises
+    ------
+    UserWarning
+        If the raw data is changed in the remote server by the author.
+    
+    Examples
+    --------
+    >>> data = QM9Dataset(label_keys=['mu', 'gap'], cutoff=5.0)
+    >>> data.num_labels
+    2
+    >>>
+    >>> # iterate over the dataset
+    >>> for g, label in data:
+    ...     R = g.ndata['R'] # get coordinates of each atom
+    ...     Z = g.ndata['Z'] # get atomic numbers of each atom
+    ...     # your code here...
+    >>>
+    """
+
+    def __init__(self,
+                 label_keys,
+                 edge_funcs=None,
+                 cutoff=5.0,
+                 raw_dir=None,
+                 force_reload=False,
+                 verbose=False):
+    
+        self.cutoff = cutoff
+        self.label_keys = label_keys
+        self.edge_funcs = edge_funcs
+        self._url = _get_dgl_url('dataset/qm9_eV.npz')
+        self._keys = ['mu', 'alpha', 'homo', 'lumo', 'gap', 'r2', 'zpve', 'U0', 'U', 'H', 'G', 'Cv']
+
+        super(QM9Dataset, self).__init__(name='qm9',
+                                         url=self._url,
+                                         raw_dir=raw_dir,
+                                         force_reload=force_reload,
+                                         verbose=verbose)
+
+    def has_cache(self):
+        """ step 1, if True, goto step 5 """
+        graph_path = f'{self.save_path}/dgl_graph.bin'
+        line_graph_path = f'{self.save_path}/dgl_line_graph.bin'
+        return os.path.exists(graph_path) and os.path.exists(line_graph_path)
+    
+    def download(self):
+        """ step 2 """
+        file_path = f'{self.raw_dir}/qm9_eV.npz'
+        if not os.path.exists(file_path):
+            download(self._url, path=file_path)
+
+    def process(self):
+        """ step 3 """
+        npz_path = f'{self.raw_dir}/qm9_eV.npz'
+        data_dict = np.load(npz_path, allow_pickle=True)
+        # data_dict['N'] contains the number of atoms in each molecule.
+        # Atomic properties (Z and R) of all molecules are concatenated as single tensors,
+        # so you need this value to select the correct atoms for each molecule.
+        self.N = data_dict['N']
+        self.R = data_dict['R']
+        self.Z = data_dict['Z']
+        self.N_cumsum = np.concatenate([[0], np.cumsum(self.N)])
+        # graph labels
+        self.label_dict = {}
+        for k in self._keys:
+            self.label_dict[k] = F.tensor(data_dict[k], dtype=F.data_type_dict['float32'])
+
+        self.label = F.stack([self.label_dict[key] for key in self.label_keys], dim=1)
+        # graph features
+        self.graphs, self.line_graphs = self._load_graph()
+    
+    def _load_graph(self):
+        num_graphs = self.label.shape[0]
+        graphs = []
+        line_graphs = []
+        
+        for idx in trange(num_graphs):
+            n_atoms = self.N[idx]
+            R = self.R[self.N_cumsum[idx]:self.N_cumsum[idx + 1]]
```

Comment:
Add a comment explaining what is `R`, what's its shape and what does each dimension stand for.